encoder/src/main/java/openlr/encoder/routesearch/SecondShortestRouteChecker.java

```diff
@@ -0,0 +1,192 @@
+            Set<Long> closedSet = new HashSet<>();
+            closedSet.add(location.get(index).getID());
+            RouteSearchData data = new RouteSearchData();
+            Line parentLine = location.get(index - 1);
+            int lengthCoveredByParentLine = lengthAlongLocation(index - 1);
+            int heuristics = lengthCoveredByParentLine + (int) GeometryUtils.distance(parentLine.getEndNode().getGeoCoordinates(),
+                    destination);
+            PQElem parent = new PQElem(parentLine, heuristics, lengthCoveredByParentLine, null);
+            data.addToOpen(parent);
+            return exploreNetworkForSecondShortestRouteUnderThreshold(closedSet, data, index);
+        } else {
+            return false;
+        }
+    }
+
+
+    /**
+     *
+     * @param current parent road segment
+     * @param closedSet Set of road segments to skip
+     * @return list of successor road segments
+     */
+    private List<Line> getAcceptableSuccessors(PQElem current, Set<Long> closedSet) {
+        List<Line> lines = new ArrayList<>();
+        current.getLine().getNextLines().forEachRemaining(lines::add);
+        return lines.stream()
+                .filter(line -> !closedSet.contains(line.getID()))
+                .filter(line -> {
+                    int newDist = current.getSecondVal() + line.getLineLength() + (int) GeometryUtils.distance(line.getEndNode().getGeoCoordinates(), destination);
+                    return networkLengthFilter(newDist);
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     *
+     * @param data priority queue list
+     * @param children list of successor road segment
+     * @param parent parent road segment
+     */
+    private void updateRouteSearchData(RouteSearchData data, List<Line> children, PQElem parent) {
+        for (Line child : children) {
+
+            int distanceCoveredByChild = parent.getSecondVal() + child.getLineLength();
+            int heuristics = distanceCoveredByChild + (int) GeometryUtils.distance(child.getEndNode().getGeoCoordinates(),
+                    destination);
+
+            if (data.hasLengthValue(child)) {
+
+                if (data.getLengthValue(child) < distanceCoveredByChild) {
+                    PQElem newElem = new PQElem(child, heuristics, distanceCoveredByChild, parent);
+                    data.updateInOpen(newElem);
+                }
+            } else {
+                data.addToOpen(new PQElem(child, heuristics, distanceCoveredByChild, parent));
+
+            }
+        }
+    }
+
+    /**
+     * length of the alternate path
+     * @param destination road segment where the alternate path join back to the actual location
+     * @param index index of the road segment where alternate path starts
+     * @return length of the alternate bathe between start and route segments of alternate path.
+     */
+    private int lengthOfSecondShortestRoute(final PQElem destination, final int index) {
```

Comment:
you can rename this method to **calculateLengthOfSecondShortestRoute**

Reply:
ac5407dc43f6f89b768b92e9571ed9f04942a5d2